datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -232,97 +242,188 @@ def __call__(sshurl, target=None, target_dir=None,
+                try:
+                    ssh(["git", "-C", path, "update-server-info"])
+                except CommandError as e:
+                    lgr.error("Failed to update server info.\n"
+                              "Error: %s" % exc_str(e))
 
         if target:
             # add the sibling(s):
             if target_url is None:
                 target_url = sshurl
             if target_pushurl is None:
                 target_pushurl = sshurl
-            result_adding = AddSibling()(dataset=ds,
-                                         name=target,
-                                         url=target_url,
-                                         pushurl=target_pushurl,
-                                         recursive=recursive,
-                                         force=existing in {'replace'})
+            AddSibling()(dataset=ds,
+                         name=target,
+                         url=target_url,
+                         pushurl=target_pushurl,
+                         recursive=recursive,
+                         force=existing in {'replace'})
 
         # TODO: Return value!?
         #       => [(Dataset, fetch_url)]
+
+    @staticmethod
+    def init_remote_repo(path, ssh, shared, dataset):
+        cmd = ["git", "-C", path, "init"]
+        if shared:
+            cmd.append("--shared=%s" % shared)
+        try:
+            ssh(cmd)
+        except CommandError as e:
+            lgr.error("Initialization of remote git repository failed at %s."
+                      "\nError: %s\nSkipping ..." % (path, exc_str(e)))
+            return False
+
+        if isinstance(dataset.repo, AnnexRepo):
+            # init remote git annex repo (part fix of #463)
+            try:
+                ssh(["git", "-C", path, "annex", "init"])
+            except CommandError as e:
+                lgr.error("Initialization of remote git annex repository failed at %s."
+                          "\nError: %s\nSkipping ..." % (path, exc_str(e)))
+                return False
+        return True
+
+    @staticmethod
+    def get_remote_git_version(ssh):
+        try:
+            out, err = ssh(["git", "version"])
+            assert out.strip().startswith("git version")
+            git_version = out.strip().split()[2]
+            lgr.debug("Detected git version on server: %s" % git_version)
+            return LooseVersion(git_version)
+
+        except CommandError as e:
+            lgr.warning(
+                "Failed to determine git version on remote.\n"
+                "Error: {0}\nTrying to configure anyway "
+                "...".format(exc_str(e)))
+        return None
+
+    @staticmethod
+    def create_postupdate_hook(path, ssh, dataset):
+        # location of post-update hook file, logs folder on remote target
+        hook_remote_target = opj(path, '.git', 'hooks', 'post-update')
+        logs_remote_target = opj(path, '.git', 'datalad', 'logs')
```

Comment:
For any datalad paths, better define them in consts.py we have so we could see all the paths we care about in one place